### PR TITLE
Refactor Navbar

### PR DESCRIPTION
**Pull request by Arthelon**
Addresses #21

### CODE REVIEW

**Comment by kiootic** [comment]
Is it possible to merge the `MobileMenu` and `Nav` components together, so the links and `<nav>` elements would not appears on the page HTML twice?

**Comment by Arthelon** [comment]
@kiootic yes this is possible. Would have to make the styles targeting those links global though since styles from the parent do not cascade to the child.

**Comment by kiootic** [comment]
~There is still two nav elements in the exported HTML. My main concern is that having two nav elements may confuse search engine.~

EDIT: my bad, was looking at old code.

File: src/components/MobileMenu.svelte [hunk 1]
```diff
@@ -1,14 +1,18 @@
+<script>
+  import NavLinks from "./NavLinks.svelte";
+</script>
+
 <style>
-  #mobile-menu-list li {
+  :global(#mobile-menu-list li) {
```
Comment (kiootic) [inline src/components/MobileMenu.svelte]: Using `:global` selector seems a rather hacky way to work around Svelte's compiler. Can we avoid it by merging the nav components together? This is how I imagine doing it:
- A `header` element containing:
    - the hamburger menu `input` & `span`s
    - the logo `img`
    - a `nav` element
        - the links `ul`
        - the action button as a separate `a` element
- On desktop
    - hamburger menu is hidden
- On mobile
    - hamburger menu is shown
    - links `ul` is switched to `position: fixed` and respond to state of `input`.

Most likely these changes is non-trivial, so we can hold it and see if we have time to do it.

File: src/components/Nav.svelte [hunk 2]
```diff
@@ -88,64 +172,102 @@
...
 
-    nav li:not(.action-item) {
-      display: none;
+    .nav__mobile-menu {
+      position: fixed;
+      top: 20px;
+      left: 8px;
+      z-index: 2;
+      cursor: pointer;
+      user-select: none;
```
Comment (kiootic) [inline src/components/Nav.svelte]: Is these two styles needed? I'm getting tap effect for the whole nav menu.

Comment (Arthelon) [reply src/components/Nav.svelte]: nope, good catch. This was from the old mobile styles.

**Review by kiootic — APPROVED** [review]

**Pull request merged**